src/ranged.cpp

```diff
@@ -247,7 +247,6 @@ dealt_projectile_attack Creature::projectile_attack( const projectile &proj_arg,
             // Don't do anything, especially don't call map::shoot as this would damage the vehicle
         } else {
             g->m.shoot( tp, proj, !no_item_damage && tp == target_arg );
-            has_momentum = proj.impact.total_damage() > 0;
```

Comment:
I'm going to revert this change, as it doesn't seem to have anything to do with the rest of the change, and I'm pretty sure it's going to break things, in particular it will allow bullets to fly through an unlimited number of walls as long as they can penetrate one.